src_c/image.c

```diff
@@ -1715,106 +1724,26 @@ MODINIT_DEFINE(image)
     if (module == NULL) {
         MODINIT_ERROR;
     }
-    st = GETSTATE(module);
 
     /* try to get extended formats */
     extmodule = PyImport_ImportModule(IMPPREFIX "imageext");
     if (extmodule) {
-        PyObject *extload;
-        PyObject *extsave;
-        PyObject *sdlImageV;
-
-        extload = PyObject_GetAttrString(extmodule, "load_extended");
-        if (!extload) {
-            Py_DECREF(extmodule);
-            DECREF_MOD(module);
-            MODINIT_ERROR;
-        }
-        extsave = PyObject_GetAttrString(extmodule, "save_extended");
-        if (!extsave) {
-            Py_DECREF(extload);
-            Py_DECREF(extmodule);
-            DECREF_MOD(module);
-            MODINIT_ERROR;
-        }
-        sdlImageV = PyObject_GetAttrString(extmodule,
-                "_get_sdl_image_version");
-        if (!sdlImageV) {
-            Py_DECREF(extload);
-            Py_DECREF(extsave);
-            Py_DECREF(extmodule);
-            DECREF_MOD(module);
-            MODINIT_ERROR;
-        }
-        if (PyModule_AddObject(module, "load_extended", extload)) {
-            Py_DECREF(extload);
-            Py_DECREF(extsave);
-            Py_DECREF(sdlImageV);
-            Py_DECREF(extmodule);
-            DECREF_MOD(module);
-            MODINIT_ERROR;
-        }
-        if (PyModule_AddObject(module, "save_extended", extsave)) {
-            Py_DECREF(extsave);
-            Py_DECREF(sdlImageV);
-            Py_DECREF(extmodule);
+        extloadobj = PyObject_GetAttrString(extmodule, "load_extended");
+        if (!extloadobj) {
             DECREF_MOD(module);
             MODINIT_ERROR;
         }
-        Py_INCREF(extload);
-        if (PyModule_AddObject(module, "load", extload)) {
-            Py_DECREF(extload);
-            Py_DECREF(sdlImageV);
-            Py_DECREF(extmodule);
+        extsaveobj = PyObject_GetAttrString(extmodule, "save_extended");
+        if (!extsaveobj) {
             DECREF_MOD(module);
             MODINIT_ERROR;
         }
-        if (PyModule_AddObject(module, "get_sdl_image_version", sdlImageV)) {
-            Py_DECREF(sdlImageV);
-            Py_DECREF(extmodule);
+        extverobj = PyObject_GetAttrString(extmodule, "_get_sdl_image_version");
```

Comment:
I think this block of changes introduces some references counting issues? When there's an error it doesn't decrement some of the objects it has made from using PyObject_GetAttrString.

Reply:
Well, I’m kind of a newbie, so never really checked about ref-count issues. I will look into it now

Reply:
My idea when I wrote that block is that, if an error occurred, then that variable is set to NULL, which is useful for later checks on wether the extended module is available or not, my question is should we decrement the refcount of a NULL pointer.

Reply:
I fixed the issue :)